### ISSUE TITLE
Support ASCII: style hs.menubar:setIcon() in hs.menubar:setMenu() menu table

### ISSUE THREAD

**Issue by bviefhues** [Hammerspoon/hammerspoon#2659]
`hs.menubar.setIcon()` supports "a string beginning with ASCII: which signifies that the rest of the string is interpreted as a special form of ASCII diagram, which will be rendered to an image and used as the icon." This is especially useful since it takes care of light mode vs dark mode rendering.

`hs.menubar.setIcon()` does support images for menu items in a menu table. However, for ASCII images, this requires piping though `hs.image.imageFromASCII()`. And that requires a light/dark mode specific`context` to adapt image to light or dark mode.

Would be great if menu table would support the same icon functionality as `setIcon()` does.

**Comment by asmagill** [comment]
I assume you mean for individual menu items within the menu?  That's a little harder than modifying `new` and `setIcon`... in the later case, we can easily wrap the functions on the lua side and easily check the image argument and if it's a string pass it through `hs.image.fromASCII` for you... in the case of the menu table, it's passed through unmodified (or returned by a function) and isn't actually parsed until the menu itself opens... it's a good idea, though, so I'll give it some thought and see what might be the best way to achieve it.

**Comment by bviefhues** [comment]
@asmagill Indeed, for individual menu items within the menu. My use case is showing a mode-dependent icon on the menubar and showing the modes as menu items, with icons attached.

I found a workaround: render the icon on the menubar and grab the image.

```
local function iconToImage(icon)
    local menubar = hs.menubar:new(false):setIcon(icon)
    local image = menubar:icon()
    menubar:delete()
    return image
end
```

The returned image fits to the light/dark mode OS setting.
